src/main/java/firebase/sdk/dashboard/dao/SDKDaoDatastore.java

```diff
@@ -0,0 +1,343 @@
+    Key sdkKey = KeyFactory.createKey("SDK", sdk.platform().toString() + "_" + sdk.libraryName());
+
+    List<EmbeddedEntity> versionMetadatas = createEmbeddedEntityVersionList(sdk.versionHistory());
+
+    Entity sdkEntity = new Entity(sdkKey);
+    sdkEntity.setProperty("platform", sdk.platform().getValue());
+    sdkEntity.setProperty("libraryName", sdk.libraryName());
+    sdkEntity.setProperty("libraryGroup", sdk.libraryGroup());
+    sdkEntity.setProperty("externalName", sdk.externalName());
+    sdkEntity.setProperty("fireEscapeName", sdk.fireEscapeName());
+    sdkEntity.setProperty("owner", sdk.owner());
+    sdkEntity.setProperty("versionHistory", versionMetadatas);
+
+    return sdkEntity;
+  }
+
+  private EmbeddedEntity createVersionMetadataEmbeddedEntity(VersionMetadata version) {
+    EmbeddedEntity metadata = new EmbeddedEntity();
+    metadata.setProperty("platform", version.platform().getValue());
+    metadata.setProperty("libraryName", version.libraryName());
+    metadata.setProperty("releaseName", version.releaseName());
+    metadata.setProperty("version", version.version());
+    metadata.setProperty("launchDate", version.launchDate().toEpochMilli());
+    return metadata;
+  }
+
+  private Entity createSDKReleaseMetadataEntity(SDKReleaseMetadata sdkReleaseMetadata) {
+    Key sdkReleaseMetadataKey = KeyFactory.createKey("SDKReleaseMetadata", sdkReleaseMetadata.platform().toString() + "_" + sdkReleaseMetadata.libraryName() + "_" + sdkReleaseMetadata.releaseName());
+
+    Entity sdkReleaseEntity = new Entity(sdkReleaseMetadataKey);
+    sdkReleaseEntity.setProperty("platform", sdkReleaseMetadata.platform().getValue());
+    sdkReleaseEntity.setProperty("libraryName", sdkReleaseMetadata.libraryName());
+    sdkReleaseEntity.setProperty("releaseName", sdkReleaseMetadata.releaseName());
+    sdkReleaseEntity.setProperty("releaseVersion", sdkReleaseMetadata.releaseVersion());
+    sdkReleaseEntity.setProperty("oldVersion", sdkReleaseMetadata.oldVersion());
+    sdkReleaseEntity.setProperty("additionalInfo", createAdditionalInfoEmbeddedEntity(sdkReleaseMetadata.additionalInfo()));
+
+    return sdkReleaseEntity;
+  }
+
+  private EmbeddedEntity createAdditionalInfoEmbeddedEntity(HashMap<String, String> additionalInfo) {
+    EmbeddedEntity entity = new EmbeddedEntity();
+    for (String key : additionalInfo.keySet()) { 
+        entity.setProperty(key, additionalInfo.get(key));
+    }
+    return entity;
+  }
+
+  private FilterPredicate makePropertyFilter(String property, Object value) {
+    FilterPredicate propertyFilter =
+        new FilterPredicate(property, FilterOperator.EQUAL, value);
+
+    return propertyFilter;
+  }
+
+  private List<VersionMetadata> getVersionHistoryFromProperty(List<EmbeddedEntity> property) {
+    if (property == null) {
+      return Arrays.asList();
+    }
+
+    ArrayList<VersionMetadata> versionHistory = new ArrayList<>();
+    for (EmbeddedEntity entity: property) {
```

Comment:
Blank space is required on both sides of the colon:
https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace